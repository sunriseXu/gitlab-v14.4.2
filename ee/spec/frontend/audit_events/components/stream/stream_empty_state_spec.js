import { shallowMount } from '@vue/test-utils';
import { GlButton } from '@gitlab/ui';
import StreamEmptyState from 'ee/audit_events/components/stream/stream_empty_state.vue';
import { mockSvgPath } from '../../mock_data';

describe('StreamEmptyState', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(StreamEmptyState, {
      provide: {
        emptyStateSvgPath: mockSvgPath,
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should emit add event', () => {
    wrapper.findComponent(GlButton).vm.$emit('click');

    expect(wrapper.emitted('add')).toBeDefined();
  });
});
