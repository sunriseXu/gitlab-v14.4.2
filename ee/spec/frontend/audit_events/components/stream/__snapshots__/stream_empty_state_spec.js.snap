// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreamEmptyState should render correctly 1`] = `
<gl-empty-state-stub
  class="gl-mt-5"
  invertindarkmode="true"
  svgheight="72"
  svgpath="mock/path"
  title="Setup streaming for audit events"
>
  <h3
    class="h4 gl-font-size-h-display gl-line-height-36 gl-mt-n3"
  >
    
      Setup streaming for audit events
    
  </h3>
  <p>
    Add an HTTP endpoint to manage audit logs in third-party systems.
  </p>
   
  <p>
    This is great for keeping everything one place.
  </p>
  <gl-button-stub
    buttontextclasses=""
    category="primary"
    icon=""
    size="medium"
    variant="confirm"
  >
    
      Add streaming destination
    
  </gl-button-stub>
</gl-empty-state-stub>
`;
